### ISSUE TITLE
Approve api AadHttpClient

### ISSUE THREAD

**Issue by kezack** [SharePoint/sp-dev-docs#1382]
**Category**
- [ ] Question
- [ ] Typo
- [x] Bug
- [ ] Additional article idea

**Expected or Desired Behavior**

I create a new webpart SPFx 1.4.1, I added the rights on package package-solution.json and package the solution.

![image](https://user-images.githubusercontent.com/2874315/36434209-cbfa391a-165e-11e8-922a-bfeca1236f87.png)

I deploy my application in the catalog app.

When approving the rights in api management, I have this error.

![image](https://user-images.githubusercontent.com/2874315/36434277-f866e9da-165e-11e8-8d6b-287122651cb6.png)

Do you have any solutions ?

**Comment by dgaeta** [comment]
@kezack Do you currently have any Approved permissions? (They would show up in the second part of the list in your screenshot)

**Comment by kezack** [comment]
@dgaeta yes, i am SharePoint Admin

**Comment by dgaeta** [comment]
Are any of the currently approved permissions a different "Microsoft Graph" permission? (i.e. the Name of the API is "Microsoft Graph" and the Permission says something else other than "Files.Read")

**Comment by kezack** [comment]
@dgaeta no, there is only one permissions "Microsoft Graph => Files.Read".

**Comment by waldekmastykarz** [comment]
@kezack just to eliminate that's it not an issue with your tenant: could you try approve the permission request either using the SPO PowerShell or the Office 365 CLI?

**Comment by kezack** [comment]
@waldekmastykarz i use Approve-SPOTenantServicePrincipalPermissionRequest and the ressource "Microsoft Graph" is not reconized.

![image](https://user-images.githubusercontent.com/2874315/36481933-bde17cb8-1711-11e8-961f-f44024646cd7.png)

**Comment by waldekmastykarz** [comment]
Could you check in your Azure AD > Enterprise applications if you have an app named GraphAggregatorService in there? (be sure to set the filter to include all apps)

**Comment by kezack** [comment]
yes, this app are here

![image](https://user-images.githubusercontent.com/2874315/36492232-f1c72152-172c-11e8-817c-87987d806839.png)

**Comment by dgaeta** [comment]
@kezack do you mind screen shotting your entire list from the Web UI page? (that includes the pending requests and the approved requests?) I am working in parallel to repro the bug but to avoid confusion it would be helpful to see your entire list.

**Comment by dgaeta** [comment]
@kezack Does your tenant have a service principal for Microsoft Graph ?

**Comment by kezack** [comment]
@dgaeta How do I know when a service Microsoft Graph in the tenant ?

**Comment by dgaeta** [comment]
@kezack are you the tenant admin?

**Comment by kezack** [comment]
@dgaeta yes

**Comment by dgaeta** [comment]
@kezack Thanks. So there are powershell commands we can use to manage service principals in the tenant.

1. Set up MSOnline V1 PowerShell module for Azure AD 
    - Install the Azure active Directory Connection library from 
      - http://connect.microsoft.com/site1164/Downloads/DownloadDetails.aspx?DownloadID=59185 
      - Select the AdministrationConfig-V1.166.0-GA.msi (be sure to select the last GA version)  

2. Open PowerShell 
3. Check if your tenant has a Microsoft Graph service principal 
    - Powershell command: `Get-MsolServicePrincipal -AppPrincipalId <Guid>`
    - `<Guid>` is the guid for Microsoft Graph in this case `00000003-0000-0000-c000-000000000000`

**Comment by kezack** [comment]
yes.

![image](https://user-images.githubusercontent.com/2874315/36501897-b47ed6fc-1748-11e8-89b4-b85227caf629.png)

**Comment by dgaeta** [comment]
@kezack Do you mind running the following command in powershell and letting me know the result?
`Get-MsolServicePrincipal -ServicePrincipalName "Microsoft Graph"`

**Comment by kezack** [comment]
@dgaeta no problem, the result in screen 
![image](https://user-images.githubusercontent.com/2874315/36508653-94b3bdaa-175d-11e8-9f97-4b36ec87b692.png)

**Comment by dgaeta** [comment]
Thanks @kezack we are looking into the issue.

**Comment by kezack** [comment]
@dgaeta did you find a solution ?

**Comment by dgaeta** [comment]
Try the following in your package-solution.json

```
"webApiPermissionRequests": [
      {
        "resource": "Microsoft.Azure.AgregatorService",
        "scope": "User.ReadBasic.All"
      }
    ]
```
Duplicate of https://github.com/SharePoint/sp-dev-docs/issues/1401

**Comment by semekhlouf** [comment]
have you solved this ? i've the same problem when i call an Azure function from SPFX
